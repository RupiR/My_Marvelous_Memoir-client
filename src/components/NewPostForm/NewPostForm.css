.NewPostForm {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  border: solid 2px #6a7374;
  padding: 20px;
}
