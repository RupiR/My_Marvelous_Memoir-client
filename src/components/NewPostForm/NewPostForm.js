import React, { Component } from "react";
import PostContext from "../../contexts/PostContext";
import PostApiService from "../../services/post-api-service";
import { Button, Textarea } from "../Utils/Utils";


export default class NewPostForm extends Component {
    static contextType = PostContext;

    handleSubmit = ev => {
        ev.preventDefault();
        const { post } = this.context;
        const { text } = ev.target;
        PostApiService.postPost(post.id, text.value)
            .then(this.context.addPost)
            .then(() => {
                text.value = "";
            })
            .catch(this.context.setError);
    };

    render() {
        return (
            <form className="NewPostForm" onSubmit={this.handleSubmit}>
                <div class="form-section">
                    <label for="entry-title">Entry title</label>
                    <input type="text" name="entry-title" placeholder="Feeling happy" required />
                </div>

                <div class="form-section">
                    <label for="entry-summary">Entry</label>
                    <textarea name="entry-summary" rows="15" required></textarea>
                </div>

                <div className='currentmood'>
                    <label htmlFor='NewPostForm__currentmood'>
                        Current Mood
                    </label>
                    <Input
                        name='Current Mood'
                        type='text'
                        required
                        id='NewPostForm__currentmood'>
                    </Input>
                </div>

                <p>Select private or public</p>

                <input type="radio" name="entry-type" value="0" class="entry-type-radio" checked />
                <label for="entry-type">
                    <span>Private</span>
                    <div class="entry-type-explanation">I would like this post to be private</div>
                </label>

                <input type="radio" name="entry-type" value="1" class="entry-type-radio" />
                <label for="entry-type">
                    <span>Public</span>
                    <div class="entry-type-explanation">I would like this post to be public</div>
                </label>

                <div className="text">
                    <Textarea
                        required
                        aria-label="Type a Post..."
                        name="text"
                        id="text"
                        cols="30"
                        rows="3"
                        placeholder="Type a Post.."
                    ></Textarea>
                </div>
                <Button type="submit">Post Post</Button>
            </form>
        );
    }
}
