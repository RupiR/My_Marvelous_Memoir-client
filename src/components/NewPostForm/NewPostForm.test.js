import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import NewPostForm from './NewPostForm';

describe('<NewPostForm />', () => {
    it('Renders without crashing', () => {
        shallow(<NewPostForm />);
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(<NewPostForm />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});