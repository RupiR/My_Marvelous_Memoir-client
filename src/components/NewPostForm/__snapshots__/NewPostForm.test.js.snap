// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewPostForm /> renders correctly 1`] = `
<form
  className="NewPostForm"
  onSubmit={[Function]}
>
  <div
    className="form-section"
  >
    <label
      htmlFor="entrytitle"
    >
      Entry title
    </label>
    <input
      name="entrytitle"
      placeholder="Feeling happy"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-section"
  >
    <label
      htmlFor="entrysummary"
    >
      Entry
    </label>
    <textarea
      name="entrysummary"
      required={true}
      rows="15"
    />
  </div>
  <p>
    Select private or public
  </p>
  <input
    className="entry-type-radio"
    defaultChecked={true}
    name="entrytype"
    type="radio"
    value="0"
  />
  <label
    htmlFor="entrytype"
  >
    <span>
      Private
    </span>
    <div
      className="entrytypeexplanation"
    >
      I would like this post to be private
    </div>
  </label>
  <input
    className="entry-type-radio"
    name="entrytype"
    type="radio"
    value="1"
  />
  <label
    htmlFor="entrytype"
  >
    <span>
      Public
    </span>
    <div
      className="entrytypeexplanation"
    >
      I would like this post to be public
    </div>
  </label>
  <button
    className="Button "
    type="submit"
  >
    Post
  </button>
</form>
`;
